src/Documentation/implementation/streams_implementation/azure_queue_streams.md

```diff
@@ -53,3 +53,62 @@ hostBuilder
     options.QueueNames = new List<string> { "yourprefix-azurequeueprovider-0" };
   }))
 ```
+
+## Tuning
+
+In a production system can emerge the need of tuning over the default configuration. When tuning there are some factors that should be considered, and it's service specific.
+
+1. First, most of the settings are per queue, so for a large number of streams, the load on each queue can be reduced by configuring more queues.
+2. Since streams process messages in order per stream, the gating factor will be the number of events being sent on a single stream.
+3. A reasonable balance of **cache time** (`StreamPullingAgentOptions.GetQueueMsgsTimerPeriod`) vs **visibility time** (`AzureQueueOptions.MessageVisibilityTimeout`) is that the visibility should be configured **to double** the time messages **are expected** to be in the cache.
+
+### Example
+
+We assume a system with this characteristics:
```

Comment:
Minor nit pick about wording here.  Can we start with 'Assuming" rather then "We assume".  I make this suggestion because a reader not being attentive to the fact that this is an assumption within an example may interpret this as being the assumptions we (the Orleans team) are making about their service, rather than the assumptions for a specific example.  "Assuming", IMO, makes it more clear that the analysis below is based on the assumption.

Reply:
I agree, much better with your suggestion!